python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -1780,28 +1822,32 @@ def do_evaluate(
         if how == "Cross":
             # Separate implementation, since cross_join returns the
             # result, not the gather maps
-            columns = plc.join.cross_join(left.table, right.table).columns()
-            left_cols = [
-                Column(new, name=old.name, dtype=old.dtype).sorted_like(old)
-                for new, old in zip(
-                    columns[: left.num_columns], left.columns, strict=True
-                )
-            ]
-            right_cols = [
-                Column(
-                    new,
-                    name=name
+            if right.num_rows == 0:
+                left_cols = Join._build_columns([], left.columns, empty=True)
+                right_cols = Join._build_columns(
+                    [],
+                    right.columns,
+                    left=False,
+                    empty=True,
+                    rename=lambda name: name
```

Comment:
IIUC, the default `rename` is `lambda name: name`, so this should be unnecessary. Am I missing something?

Reply:
And FWIW, it seems like we could *maybe* pass in the list of renamed values, rather than a callable to produce it.

In the simple case, it should just be

```
renamed_columns = list(left.columns)
```

And in the more complex case below, I think its

```
renamed_columns = [
    name
    if name not in left.column_names_set
    else f"{name}{suffix}"
    for column in right.columns
]
```


This would make this function a bit simpler to serialize (though I don't know whether or not we serialize this function with the streaming executor, so we should only do this if we think it's an improvement).